storage/rocksdb/ha_rocksdb.cc

```diff
@@ -6327,17 +6343,23 @@ int ha_rocksdb::read_row_from_secondary_key(uchar *const buf,
   int rc = 0;
   uint pk_size;
 
-  if (m_keyread_only && m_lock_rows == RDB_LOCK_NONE && !has_hidden_pk(table)) {
-    /* Get the key columns and primary key value */
-    const rocksdb::Slice &rkey = m_scan_it->key();
+  /* Get the key columns and primary key value */
+  const rocksdb::Slice &rkey = m_scan_it->key();
+  const rocksdb::Slice &value = m_scan_it->value();
+
+  DBUG_ASSERT_IMP(!m_keyread_only, !m_lookup_bitmap_vect.empty());
```

Comment:
ha_rocksdb::load_auto_incr_value will use secondary_index_read without a call to index_init, so the vector will be empty, but sets m_keyread_only to true before doing so. So in that specific case, it's fine that the vector is empty. I also put the assert in read_row_from_secondary_key as a sanity check, but it's probably not needed since it won't be called like that.